lib/ui/window.dart

```diff
@@ -150,7 +150,37 @@ class Locale {
   /// Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
   /// Typically this means the primary language subtag should be lowercase and
   /// the region subtag should be uppercase.
-  const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
+  const Locale(
+    this._languageCode, [
+    this._countryCode,
+  ]) : assert(_languageCode != null),
+       this.scriptCode = null,
+       this._variants = null;
+
+  /// Creates a new Locale object.
+  ///
+  /// The keyword arguments specify the subtags of the Locale. The subtag values
+  /// should match the values listed as valid in Unicode CLDR supplemental data:
+  /// [language](http://unicode.org/cldr/latest/common/validity/language.xml),
+  /// [script](http://unicode.org/cldr/latest/common/validity/script.xml),
+  /// [region](http://unicode.org/cldr/latest/common/validity/region.xml) and
+  /// [variant](http://unicode.org/cldr/latest/common/validity/variant.xml).
+  /// There may be multiple variants as long as they appear in sorted order and
+  /// are joined by hyphens.
+  const Locale.fromComponents({
```

Comment:
Discussion point: what's the best name for this constructor?

Reply:
Perhaps just "create()" ?
"components" is like "data" it's a bit of a non-word as far as getting meaning across.
I think the 4 arguments of the function are likely to be enough of a "forcing function" to make people look at the docs to figure out what's going on (this doesn't seem like one of those "convenience" methods, but rather the "canonical method"). I'd expect helpers like "fromLanguage()" and so on to come later.